### PR TITLE
executor: validate snapshot TS

### PR DESCRIPTION
**Pull request by coocood**
Fixes https://github.com/pingcap/tidb/issues/2485

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

File: executor/set.go [hunk 1]
```diff
@@ -123,10 +126,19 @@ func (e *SetExecutor) executeSet() error {
 			if err != nil {
 				return errors.Trace(err)
 			}
+			olsSnapshotTS := sessionVars.SnapshotTS
```
Comment (tiancaiamao) [inline executor/set.go]: oldSnapshotTS ?

File: executor/set.go [hunk 2]
```diff
@@ -123,10 +126,19 @@ func (e *SetExecutor) executeSet() error {
...
 			err = varsutil.SetSessionSystemVar(sessionVars, name, value)
 			if err != nil {
 				return errors.Trace(err)
 			}
+			newSnapshotIsSet := sessionVars.SnapshotTS > 0 && sessionVars.SnapshotTS != olsSnapshotTS
```
Comment (tiancaiamao) [inline executor/set.go]: Why not this logic before line 129?

Comment (coocood) [inline executor/set.go]: before It's a string value, after it's a uint64. it's easier to compare.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**